File: grph/grph.rb
require_relative '../lib/rosalind.rb'
include Rosalind

dnas = parse_fasta_file ARGV[0]
ids = dnas.keys

adjacents = []

ids.each_index do |i|
  ids.each_index do |j|
    next if i==j
    if is_adjacent?(dnas[ids[i]], dnas[ids[j]], 3)
      adjacents << ids[i] + ' ' + ids[j]
    end
  end
end

output = adjacents.join("\n")
puts output
write_file output

File: python/grph/grph.py
import sys
from os import path
#load the libraries
sys.path.insert(0, path.abspath(path.join(path.dirname(__file__), '../lib')))
import rosalind_util as util

from methods import is_adjacent

dna_db = util.read_fasta_file(str(sys.argv[1]))

output =''
for id1 in dna_db.keys():
	for id2 in dna_db.keys():
		if id1==id2: continue
		if is_adjacent(dna_db[id1], dna_db[id2]):
			output += ' '.join([id1, id2]) + '\n'


util.write_file(output)

